kafka.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"context"
+
+	"github.com/segmentio/kafka-go"
+	log "github.com/sirupsen/logrus"
+)
+
+// ProducerConfig configures the producer with needed kafka information
+type ProducerConfig struct {
```

Comment:
I go back and forth whether I like this paradigm of creating "Config" structs. More often than not, instances of the "Config" struct are used once as an argument to the factory function. So we don't gain much over simply passing the necessary arguments directly into the factory function.